src/endpoint/s3/ops/s3_get_bucket.js

```diff
@@ -70,7 +70,7 @@ async function get_bucket(req) {
                     reply.next_marker, reply.objects, reply.is_truncated),
             } : { // list_type v1
                 Marker: req.query.marker || '',
-                NextMarker: req.query.delimiter ? reply.next_marker : undefined,
+                NextMarker: reply.next_marker?.marker_pos ? JSON.stringify(reply.next_marker) : reply.next_marker,
```

Comment:
I don't think we can change that for list_objects v1 at all, this will break compatibility.
We should do that only for `NextContinuationToken` in list_type === '2' above.